pkg/manager/router/backend_selector.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package router
+
+type BackendSelector struct {
+	excluded  []string
+	cur       string
+	routeOnce func(excluded []string) string
+	addConn   func(addr string, conn RedirectableConn) error
+}
+
+func (bs *BackendSelector) Reset() {
```

Comment:
I think we can just delete `Reset()` API, making selector always returning a possible addr until we timeout, or we call `Succeed`.

Reply:
I'm afraid it will be not easy to control backoff. Say, I want to iterate all the backends first, and then sleep for 100ms before continuing the next loop, and then sleep for 200ms before the next.

Reply:
Makes sense.